File: backends/php-laravel/app/Domain/Notification/Condition/NotificationCondition.php
<?php

/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */

declare(strict_types=1);

namespace App\Domain\Notification\Condition;

use App\Support\Repositories\Conditions\AbstractCondition;

/**
 * 通知Conditionクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 * @package App\Domain\Notification\Condition
 */
class NotificationCondition extends AbstractCondition
{
    public ?int $staffId = null;
    public ?int $messageType = null;
    public ?string $title = null;
    public ?string $message = null;
    public bool $read = false;
    public bool $countUnread = false;
    public ?string $cursor;
    public int $limit = 1;
    public array $ids = [];
    public bool $all = false;
}

File: backends/java-spring/src/main/java/com/authorization/domain/notification/condition/NotificationCondition.java
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
package com.authorization.domain.notification.condition;

import com.authorization.support.repositories.conditions.AbstractCondition;
import java.util.List;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 通知Conditionクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class NotificationCondition extends AbstractCondition {

    private Long staffId;
    private Integer messageType;
    private String title;
    private String message;
    private boolean read;
    private boolean countUnread;
    private String cursor;
    private int limit = 1;
    private List<Long> ids = List.of();
    private boolean all;
}
